api/src/main/java/org/apache/iceberg/geospatial/GeospatialPredicateEvaluators.java

```diff
@@ -0,0 +1,214 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.geospatial;
+
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+
+public class GeospatialPredicateEvaluators {
+  private GeospatialPredicateEvaluators() {}
+
+  public interface GeospatialPredicateEvaluator {
+    /**
+     * Determines whether the two bounding boxes intersect.
+     *
+     * @param bbox1 the first bounding box
+     * @param bbox2 the second bounding box
+     * @return true if this box intersects the other box
+     */
+    boolean intersects(BoundingBox bbox1, BoundingBox bbox2);
+  }
+
+  /**
+   * Create an evaluator for evaluating bounding box relationship for the given geospatial type.
+   *
+   * @param type the geospatial type, should be one of Type.TypeID.GEOMETRY or Type.TypeID.GEOGRAPHY
+   * @return the evaluator
+   */
+  public static GeospatialPredicateEvaluator create(Type type) {
+    switch (type.typeId()) {
+      case GEOMETRY:
+        return create((Types.GeometryType) type);
+      case GEOGRAPHY:
+        return create((Types.GeographyType) type);
+      default:
+        throw new UnsupportedOperationException("Unsupported type for BoundingBox: " + type);
+    }
+  }
+
+  /**
+   * Create an evaluator for evaluating bounding box relationship for planar geometries
+   *
+   * @return the evaluator
+   */
+  public static GeometryEvaluator create(Types.GeometryType type) {
+    return new GeometryEvaluator();
+  }
+
+  /**
+   * Create an evaluator for evaluating bounding box relationship for geographies
+   *
+   * @return the evaluator
+   */
+  public static GeographyEvaluator create(Types.GeographyType type) {
+    return new GeographyEvaluator();
+  }
+
+  public static class GeometryEvaluator implements GeospatialPredicateEvaluator {
+
+    /**
+     * Check if two bounding boxes intersect
+     *
+     * @param bbox1 the first bounding box
+     * @param bbox2 the second bounding box
+     * @return true if the bounding boxes intersect
+     */
+    @Override
+    public boolean intersects(BoundingBox bbox1, BoundingBox bbox2) {
+      if (!intersectsYZM(bbox1, bbox2)) {
+        return false;
+      }
+
+      // Check X dimension (longitude/easting) - no wrap-around
+      return rangeIntersects(bbox1.min().x(), bbox1.max().x(), bbox2.min().x(), bbox2.max().x());
+    }
+
+    static boolean intersectsYZM(BoundingBox bbox1, BoundingBox bbox2) {
```

Comment:
nit: this util method can be pulled outside of this nested class. it doesn't affect the code sharing with `GeographyEvaluator`. just will be a little more symmetric.